cli/democluster: TestTransientClusterSimulateLatencies failed

cli/democluster.TestTransientClusterSimulateLatencies [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9763431?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9763431?buildTab=artifacts#/) on release-22.2 @ [ac6bd23042f11dbf7e3d444a2fd7d49b35d73b50](https://github.com/cockroachdb/cockroach/commits/ac6bd23042f11dbf7e3d444a2fd7d49b35d73b50):


```
=== RUN   TestTransientClusterSimulateLatencies
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1937dc802442f15720877ed362f2998f/logTestTransientClusterSimulateLatencies477399448
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestTransientClusterSimulateLatencies
    demo_cluster_test.go:256: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1937dc802442f15720877ed362f2998f/logTestTransientClusterSimulateLatencies477399448
--- FAIL: TestTransientClusterSimulateLatencies (1.45s)
=== RUN   TestTransientClusterSimulateLatencies/from_europe-west1
    demo_cluster_test.go:247: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3024/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cli/democluster/democluster_test_/democluster_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/democluster/demo_cluster_test.go:247
        	Error:      	Should be true
        	Test:       	TestTransientClusterSimulateLatencies/from_europe-west1
        	Messages:   	expected duration at least 73ms, got 71.130619ms
    --- FAIL: TestTransientClusterSimulateLatencies/from_europe-west1 (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101282 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot branch-release-23.1.0]
- #100389 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot branch-release-23.1]
- #99768 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot T-sql-sessions branch-master skipped-test]
</p>
</details>
/cc @cockroachdb/sql-sessions @cockroachdb/server @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransientClusterSimulateLatencies.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27290

cli/democluster.TestTransientClusterSimulateLatencies [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9977691?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9977691?buildTab=artifacts#/) on release-22.2 @ [a875d58204554ea945300d183e5176bff6d4e703](https://github.com/cockroachdb/cockroach/commits/a875d58204554ea945300d183e5176bff6d4e703):


```
=== RUN   TestTransientClusterSimulateLatencies
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1937dc802442f15720877ed362f2998f/logTestTransientClusterSimulateLatencies4152040889
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestTransientClusterSimulateLatencies
    demo_cluster_test.go:256: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1937dc802442f15720877ed362f2998f/logTestTransientClusterSimulateLatencies4152040889
--- FAIL: TestTransientClusterSimulateLatencies (1.90s)
=== RUN   TestTransientClusterSimulateLatencies/from_us-west1
    demo_cluster_test.go:247: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cli/democluster/democluster_test_/democluster_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/democluster/demo_cluster_test.go:247
        	Error:      	Should be true
        	Test:       	TestTransientClusterSimulateLatencies/from_us-west1
        	Messages:   	expected duration at least 73ms, got 68.033211ms
    --- FAIL: TestTransientClusterSimulateLatencies/from_us-west1 (0.07s)
=== RUN   TestTransientClusterSimulateLatencies/from_europe-west1
    demo_cluster_test.go:247: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cli/democluster/democluster_test_/democluster_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/democluster/demo_cluster_test.go:247
        	Error:      	Should be true
        	Test:       	TestTransientClusterSimulateLatencies/from_europe-west1
        	Messages:   	expected duration at least 73ms, got 70.124622ms
    --- FAIL: TestTransientClusterSimulateLatencies/from_europe-west1 (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101282 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot branch-release-23.1.0]
- #100389 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot branch-release-23.1]
- #99768 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot T-sql-sessions branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransientClusterSimulateLatencies.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli/democluster.TestTransientClusterSimulateLatencies [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10009209?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10009209?buildTab=artifacts#/) on release-22.2 @ [9145535e8f6e57d09e3688d6b95fbdceecc47194](https://github.com/cockroachdb/cockroach/commits/9145535e8f6e57d09e3688d6b95fbdceecc47194):


```
=== RUN   TestTransientClusterSimulateLatencies
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1937dc802442f15720877ed362f2998f/logTestTransientClusterSimulateLatencies3979070802
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestTransientClusterSimulateLatencies
    demo_cluster_test.go:256: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1937dc802442f15720877ed362f2998f/logTestTransientClusterSimulateLatencies3979070802
--- FAIL: TestTransientClusterSimulateLatencies (2.77s)
=== RUN   TestTransientClusterSimulateLatencies/from_europe-west1
    demo_cluster_test.go:247: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2687/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cli/democluster/democluster_test_/democluster_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/democluster/demo_cluster_test.go:247
        	Error:      	Should be true
        	Test:       	TestTransientClusterSimulateLatencies/from_europe-west1
        	Messages:   	expected duration at least 73ms, got 67.947025ms
    --- FAIL: TestTransientClusterSimulateLatencies/from_europe-west1 (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101282 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot branch-release-23.1.0]
- #100389 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot branch-release-23.1]
- #99768 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot T-sql-sessions branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransientClusterSimulateLatencies.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli/democluster.TestTransientClusterSimulateLatencies [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10027898?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10027898?buildTab=artifacts#/) on release-22.2 @ [163c2296507466cb3fb833bfd32484a62b405460](https://github.com/cockroachdb/cockroach/commits/163c2296507466cb3fb833bfd32484a62b405460):


```
=== RUN   TestTransientClusterSimulateLatencies
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1937dc802442f15720877ed362f2998f/logTestTransientClusterSimulateLatencies192375329
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestTransientClusterSimulateLatencies
    demo_cluster_test.go:256: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1937dc802442f15720877ed362f2998f/logTestTransientClusterSimulateLatencies192375329
--- FAIL: TestTransientClusterSimulateLatencies (2.13s)
=== RUN   TestTransientClusterSimulateLatencies/from_europe-west1
    demo_cluster_test.go:247: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2879/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cli/democluster/democluster_test_/democluster_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/democluster/demo_cluster_test.go:247
        	Error:      	Should be true
        	Test:       	TestTransientClusterSimulateLatencies/from_europe-west1
        	Messages:   	expected duration at least 73ms, got 70.093064ms
    --- FAIL: TestTransientClusterSimulateLatencies/from_europe-west1 (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101282 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot branch-release-23.1.0]
- #100389 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot branch-release-23.1]
- #99768 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot T-sql-foundations branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransientClusterSimulateLatencies.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli/democluster.TestTransientClusterSimulateLatencies [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10047341?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10047341?buildTab=artifacts#/) on release-22.2 @ [97259ff3650b28abde99a8174d70016c9b5daa28](https://github.com/cockroachdb/cockroach/commits/97259ff3650b28abde99a8174d70016c9b5daa28):


```
=== RUN   TestTransientClusterSimulateLatencies
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1937dc802442f15720877ed362f2998f/logTestTransientClusterSimulateLatencies3984795248
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestTransientClusterSimulateLatencies
    demo_cluster_test.go:256: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1937dc802442f15720877ed362f2998f/logTestTransientClusterSimulateLatencies3984795248
--- FAIL: TestTransientClusterSimulateLatencies (2.39s)
=== RUN   TestTransientClusterSimulateLatencies/from_us-west1
    demo_cluster_test.go:247: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2872/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cli/democluster/democluster_test_/democluster_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/democluster/demo_cluster_test.go:247
        	Error:      	Should be true
        	Test:       	TestTransientClusterSimulateLatencies/from_us-west1
        	Messages:   	expected duration at least 73ms, got 64.205967ms
    --- FAIL: TestTransientClusterSimulateLatencies/from_us-west1 (0.06s)
=== RUN   TestTransientClusterSimulateLatencies/from_europe-west1
    demo_cluster_test.go:247: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2872/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cli/democluster/democluster_test_/democluster_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/democluster/demo_cluster_test.go:247
        	Error:      	Should be true
        	Test:       	TestTransientClusterSimulateLatencies/from_europe-west1
        	Messages:   	expected duration at least 73ms, got 56.10261ms
    --- FAIL: TestTransientClusterSimulateLatencies/from_europe-west1 (0.06s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101282 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot branch-release-23.1.0]
- #100389 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot branch-release-23.1]
- #99768 cli/democluster: TestTransientClusterSimulateLatencies failed [C-test-failure O-robot T-sql-foundations branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransientClusterSimulateLatencies.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cli/democluster/demo_cluster_test.go
+++ b/pkg/cli/democluster/demo_cluster_test.go
@@ -141,16 +141,16 @@ func TestTransientClusterSimulateLatencies(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	// This is slow under race as it starts a 9-node cluster which
+	// This is slow under race and deadlock as it starts a 9-node cluster which
 	// has a very high simulated latency between each node.
 	skip.UnderRace(t)
-
-	skip.WithIssue(t, 102115, "flaky test")
+	skip.UnderDeadlock(t)
 
 	demoCtx := newDemoCtx()
 	// Set up an empty 9-node cluster with simulated latencies.
 	demoCtx.SimulateLatency = true
 	demoCtx.NumNodes = 9
+	demoCtx.Localities = defaultLocalities
 
 	certsDir := t.TempDir()
 
@@ -186,11 +186,21 @@ func TestTransientClusterSimulateLatencies(t *testing.T) {
 		return s.RunLocalSQL(ctx,
 			func(ctx context.Context, ie *sql.InternalExecutor) error {
 				_, err := ie.Exec(
-					ctx, "admin-user", nil,
+					ctx, "create-admin-user", nil,
 					fmt.Sprintf("CREATE USER %s WITH PASSWORD $1", adminUser),
 					adminPassword,
 				)
-				return err
+				if err != nil {
+					return err
+				}
+				_, err = ie.Exec(
+					ctx, "grant-admin-user", nil,
+					fmt.Sprintf("GRANT ADMIN TO %s", adminUser),
+				)
+				if err != nil {
+					return err
+				}
+				return nil
 			})
 	}))
 

